// dirigera/dist/src/api/rooms.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.default = (got) => {
    return {
        async list() {
            return (await got.get(`rooms`).json());
        },
        async get({ id }) {
            return (await got.get(`rooms/${id}`).json());
        },
        // TODO add room management API
        //  * createRoom
        //  * deleteRoom
        //  * moveRoomDevices
        //  * updateRoom
        // TODO cover device-type specific attribute changes
        async setIsOn({ id, deviceType, isOn, }) {
            return got
                .patch(`devices/set/${id}`, {
                searchParams: {
                    deviceType,
                },
                json: [
                    {
                        attributes: {
                            isOn,
                        },
                    },
                ],
            })
                .json();
        },
        // TODO low level API, shouldn't be exposed
        async setAttributes({ id, deviceType, attributes, transitionTime, }) {
            return got
                .patch(`devices/room/${id}`, {
                searchParams: {
                    deviceType,
                },
                json: [
                    {
                        attributes,
                        transitionTime,
                    },
                ],
            })
                .json();
        },
    };
};
